model-optimizer/extensions/middle/L2NormFusing.py

```diff
@@ -86,29 +92,33 @@ def replace_pattern(self, graph: Graph, match: dict):
         # reduction only along spatial and channel dimensions.
         input_rank = len(match['sum'].in_port(0).data.get_shape())
         if input_rank not in [2, 3, 4]:
-            log.debug('IE supports L2 normalization only for 2D, 3D and 4D tensors, skip fusing transformation.')
+            log.debug('IE supports L2 normalization only for 2D, 3D and 4D tensors.')
             return
 
         axes = match['sum'].in_port(1).data.get_value()
         axes = int64_array(axes)
         if axes.shape == ():
             axes = int64_array([axes])
+        axes = int64_array([axis if axis >= 0 else axis + input_rank for axis in axes])
         axes.sort()
 
-        if not np.array_equal(axes, int64_array(np.arange(start=1, stop=input_rank))):
-            log.debug('IE doesn\'t support l2 normalization with reduction along axes {}, skip fusing transformation.'
-                      ''.format(axes))
+        transformation_applicable = False
+        # check for case C + all spatial dims. Works for 2D (NC), 3D (NCH) and 4D (NCHW and NHWC)
+        if len(axes) + 1 == input_rank and np.array_equal(axes, int64_array(np.arange(start=1, stop=input_rank))):
+            transformation_applicable = True
+
+        # check for pure C channel normalization
+        if len(axes) == 1 and ((input_rank == 4 and get_features_dim(graph.graph['layout'], input_rank) == axes[0]) or
```

Comment:
why was it not called as `get_channels_dim`?

Reply:
Due to historical reasons.